### ISSUE TITLE
Uniform stroke pushes bounds too far

### ISSUE THREAD

**Issue by kirill-konshin** [fabricjs/fabric.js#7651]
## Version
4.6.0

## Test Case

https://jsfiddle.net/dfuse/Lwpfkrx1/

## Steps to reproduce

Create a path with stroke and put it inside group, set uniform stroke to `false`. Select the group.

You can click "Toggle uniform" and "Toggle stroke width" to see different combinations.

## Expected Behavior

Selection should stick to border of stroke.

<img width="564" alt="image" src="https://user-images.githubusercontent.com/1530410/152630798-75aa3174-249e-4fb8-ac1e-7f5caaab4131.png">

## Actual Behavior

<img width="599" alt="image" src="https://user-images.githubusercontent.com/1530410/152630788-6f1a5a12-3fd9-4bd6-9277-706d206acd4a.png">

## Background

When I observed this behavior, I thought it might be useful to introduce a proper inner stroke, as mentioned here https://github.com/fabricjs/fabric.js/issues/2378 so that with uniform stroke the size of stroke won't affect any dimensions.

**Comment by asturur** [comment]
This is a known issue, this is probably going away with the group rewrite.

**Comment by asturur** [comment]
For reference https://github.com/fabricjs/fabric.js/issues/6705

**Comment by ShaMan123** [comment]
@asturur is this related to the fix of #7476?

**Comment by ShaMan123** [comment]
I see `strokeUniform` is buggy across fabric

**Comment by ShaMan123** [comment]
related #7605 ?

**Comment by ShaMan123** [comment]
@kirill-konshin did you try #7639 ?

**Comment by asturur** [comment]
Well buggy across fabric is maybe an exageration :) 
It does not work inside groups, but probably with the group rewrite this is going to change a bit.

**Comment by ShaMan123** [comment]
😳

**Comment by asturur** [comment]
> @asturur is this related to the fix of #7476?

No, but your code could use take in consideration strokeUniform, since we are not doing that.

The truth is that strokeUniform is a patch, while the real solution is probably to transform path's and polygon's points to absorbe the transformation.

strokeUniform does not apply to text either. Or if it does we are patching it manually.

Is probably safer with controlApi to build something that change the path data while you transform it, rather than having strokeUniform work consistently.

**Comment by kirill-konshin** [comment]
"transform path's and polygon's points to absorbe the transformation" I tried that, the problem is that when shape is scaled, I have to recalculate the path at least once scaling is finished, so path "jumps" to new place.

I can recalculate at 60fps, but this may not perform well...

**Comment by kirill-konshin** [comment]
Not stale. Still needs to be addressed.

**Comment by Errin890** [comment]
Any updates on this issue?

**Comment by ShaMan123** [comment]
#8040 #8344

**Comment by sawadyecma** [comment]
Hi, just wondering if there are any updates on this issue.
I’ve been trying to work around it in my project, but I’m still encountering a problem where a rect with strokeUniform: true doesn't align correctly within its parent group’s bounding box.

The reproduction: https://codesandbox.io/p/sandbox/9fchlk

I suspect that this behavior is caused by how getObjectBounds() works internally.

https://github.com/fabricjs/fabric.js/blob/d8c58db563c73f3887e70bdc59431acb6c87ba90/src/LayoutManager/LayoutStrategies/utils.ts#L16
